### PR TITLE
Update ERC-7066: Move to Last Call

### PR DESCRIPTION
**Pull request by piyush-chittara**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

**Comment by Pandapip1** [comment]
@eth-bot rerun

File: ERCS/erc-7066.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: Lockable Extension for ERC-721
 description: Interface for enabling locking of ERC-721 using locker and approved
 author: Piyush Chittara (@piyush-chittara), StreamNFT (@streamnft-tech), Srinivas Joshi (@SrinivasJoshi)
 discussions-to: https://ethereum-magicians.org/t/eip-7066-lockable-extension-for-erc721/14425
-status: Review
+status: Final
```
Comment (SamWilsn) [inline ERCS/erc-7066.md:7]: You'll need to begin a Last Call period and add a `last-call-deadline` before you can go to Final. The Last Call status gives any potential reviewers notice that the proposal is about to go to Final.

 ```suggestion
status: Last Call
last-call-deadline: 2023-11-20
```

File: ERCS/erc-7066.md [hunk 2]
```diff
@@ -13,17 +13,17 @@ requires: 165, 721
 
 ## Abstract
 
-An extension of [ERC-721](./eip-721.md), this standard incorporates `locking` features into NFTs, allowing for various uses while preventing sale or transfer. The token's `owner` can `lock` it, setting up locker address (either an EOA or a contract) that exclusively holds the power to unlock the token. Owner can also provide approval for tokenId, enabling ability to lock asset while address holds the token approval. Token can also be locked by `approved`, assigning locker to itself. Upon token transfer, these rights get purged.
+An extension of [ERC-721](./erc-721.md), this standard incorporates `locking` features into NFTs, allowing for various uses while preventing sale or transfer. The token's `owner` can `lock` it, setting up locker address (either an EOA or a contract) that exclusively holds the power to unlock the token. Owner can also provide approval for tokenId, enabling ability to lock asset while address holds the token approval. Token can also be locked by `approved`, assigning locker to itself. Upon token transfer, these rights get purged.
```
Comment (SamWilsn) [inline ERCS/erc-7066.md:16]:
```suggestion
An extension of [ERC-721](./erc-721.md), this standard incorporates `locking` features into NFTs, allowing for various uses while preventing sale or transfer. The token's `owner` can `lock` it, setting up locker address (either an EOA or a contract) that exclusively holds the power to unlock the token. Owner can also provide approval for `tokenId`, enabling ability to lock asset while address holds the token approval. Token can also be locked by `approved`, assigning locker to itself. Upon token transfer, these rights get purged.
```

File: ERCS/erc-7066.md [hunk 3]
```diff
@@ -13,17 +13,17 @@ requires: 165, 721
...
 
 ## Motivation
 
-[ERC-721](./eip-721.md) has sparked an unprecedented surge in demand for NFTs. However, despite this tremendous success, NFT economy suffers from secondary liquidity where it remains Illiquid in owner’s wallet. There are projects such as NFTfi, Paraspace which aims to address the liquidity challenge, but they entail below mentioned inconveniences and risks for owners as they necessitate transferring the participating NFTs to the projects' contracts. 
+[ERC-721](./erc-721.md) has sparked an unprecedented surge in demand for NFTs. However, despite this tremendous success, NFT economy suffers from secondary liquidity where it remains Illiquid in owner’s wallet. There are projects such as NFTfi, Paraspace which aims to address the liquidity challenge, but they entail below mentioned inconveniences and risks for owners as they necessitate transferring the participating NFTs to the projects' contracts.
```
Comment (SamWilsn) [inline ERCS/erc-7066.md:20]: Should probably avoid mentioning specific projects:

```suggestion
[ERC-721](./erc-721.md) has sparked an unprecedented surge in demand for NFTs. However, despite this tremendous success, the NFT economy suffers from secondary liquidity where it remains illiquid in owner’s wallet. There are projects which aim to address the liquidity challenge, but they entail the below mentioned inconveniences and risks for owners as they necessitate transferring the participating NFTs to the projects' contracts. 
```

File: ERCS/erc-7066.md [hunk 4]
```diff
@@ -13,17 +13,17 @@ requires: 165, 721
...
 - Lack of composability: The market could benefit from increased liquidity if NFT owners had access to multiple financial tools, such as leveraging loans and renting out their assets for maximum returns. Composability serves as the missing piece in creating a more efficient market.
 - Smart contract vulnerabilities: NFTs are susceptible to loss or theft due to potential bugs or vulnerabilities present in the smart contracts they rely on.
 
-The aforementioned issues contribute to a poor user experience (UX), and we propose enhancing the [ERC-721](./eip-721.md) standard by implementing a native locking mechanism: 
+The aforementioned issues contribute to a poor user experience (UX), and we propose enhancing the [ERC-721](./erc-721.md) standard by implementing a native locking mechanism: 
 Rather than being transferred to a smart contract, an NFT remains securely stored in self-custody but is locked. 
 During the lock period, the NFT's transfer is restricted while its other properties remain unchanged. 
 NFT Owner retains the ability to use or distribute it’s utility
```
Comment (SamWilsn) [inline ERCS/erc-7066.md:29]:
```suggestion
NFT Owner retains the ability to use or distribute it’s utility.
```

File: ERCS/erc-7066.md [hunk 5]
```diff
@@ -47,15 +47,15 @@ This proposal is different from other locking proposals in number of ways:
 - This implementation proposes a separation of rights between locker and approver. Token can be locked with approval and approved can unlock and withdraw tokens (opening up opportunities like renting, lending, bnpl etc), and token can be locked lacking the rights to revoke token, yet can unlock if required (opening up opportunities like account-bound NFTs).
```
Comment (SamWilsn) [inline ERCS/erc-7066.md:47]:
```suggestion
- This implementation proposes a separation of rights between locker and approver. Token can be locked with approval and approved can unlock and withdraw tokens (opening up opportunities like renting, lending, BNPL etc), and token can be locked lacking the rights to revoke token, yet can unlock if required (opening up opportunities like account-bound NFTs).
```

Also, please include the abbreviation on line 38:

> - Buy Now Pay Later (BNPL): ...

We prefer that abbreviations are expanded the first time they are used.

**Review by SamWilsn — COMMENTED** [review]
Line 64 says:

> `lock` function MUST revert if token is **not** already locked.

Emphasis mine. This seems backwards?

---

A bit later you write:

> ERC-721 `_tansfer` function [...]

ERC-721 doesn't have a `_tansfer` function (nor `_transfer`), so these requirements don't make sense. You may be thinking of the OpenZeppelin implementation, which does have `_transfer`.

You can't depend on a specific implementation of ERC-721 in a standard, so you should reword this to be more general. Something like:

> ERC-721 functions that transfer ownership of a token MUST revert [...]

**Comment by piyush-chittara** [comment]
Thanks a lot for the reviews @Joeysantoro @SamWilsn!
Your pointers are looked into and resolved.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**